presto-main/src/main/java/io/prestosql/testing/TestingConnectorSession.java

```diff
@@ -161,4 +151,59 @@ public String toString()
                 .omitNullValues()
                 .toString();
     }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static class Builder
+    {
+        private ConnectorIdentity identity = new ConnectorIdentity("user", Optional.empty(), Optional.empty());
+        private Optional<String> source = Optional.of("test");
```

Comment:
why optional? You cannot change it.
Also, make fields that you cannot modify finals